src/speech_node/speech.py

```diff
@@ -26,54 +26,133 @@ class bcolors:
 
 
 class TTS(object):
-    def __init__(self):
+    """
+    Bridge from TTS service calls and topic messages to Toyota TTS, introducing a buffer to handle all requests in
+    receiving order
+    """
+    def __init__(self, rate):
+        """
+        Constructor
+        :param rate: ROS parameter for spin-rate
+        """
+
         self.samples_path = rospy.get_param("~samples_path", "~/MEGA/media/audio/soundboard")
+        self.rate = rate
+
+        # buffer and goal state
+        self.buffer = collections.deque()
+        self.block_queue = False
 
         # topics
         self.sub_speak = rospy.Subscriber("~input", String, self.speak)
         # self.pub_speak = rospy.Publisher("/talk_request", Voice, queue_size=10)
 
         # services
         self.srv_speak = rospy.Service('~speak', Speak, self.speak_srv)
+        self.srv_clear_buffer = rospy.Service('~clear_buffer', Empty, self.clear_buffer_srv)
 
         # clients
         self.speech_client = actionlib.SimpleActionClient('/talk_request_action', TalkRequestAction)
         self.speech_client.wait_for_server()
 
-    def do_tts(self, req):
-        # rospy.loginfo('TTS: Toyota TTS, through bridge node. "' + bcolors.OKBLUE + req.sentence + bcolors.ENDC + '"')
+    def buffer_requests(self, req):
+        """
+        Handle requests: put them in buffer and wait if there is a blocking TTS call in the buffer
+        :param req: text_to_speech.srv server message - SpeakRequest
```

Comment:
Documenting the type can be done with 

```
:type req: text_to_speech.srv.SpeakRequest
```
But since you import the type above, you can do just 

```
:type req: SpeakRequest
```